packages/rich-text/src/component/index.js

```diff
@@ -71,7 +71,9 @@ export function useRichText( {
 	function setRecordFromProps() {
 		_value.current = value;
 		record.current = create( {
-			html: preserveWhiteSpace ? value : collapseWhiteSpace( value ),
+			html: preserveWhiteSpace
+				? value
```

Comment:
I tested with `presrveWhiteSpace` == `true` and it doesn't crash, so the problem seems limited to the `collapseWhiteSpace` call.